### ISSUE TITLE
v-bind="$attrs" and v-on="$listeners" does not work when using <component :is="">

### ISSUE THREAD

**Issue by sybrendotinga** [vuejs/vue-test-utils#486]
### Version
1.0.0-beta.12

### Reproduction link
[https://codesandbox.io/s/j3p792o4jy](https://codesandbox.io/s/j3p792o4jy)

### Steps to reproduce
1. Go to the provided CodeSandbox (https://codesandbox.io/s/j3p792o4jy)
2. Check the component, it renders the currency component succesfully
3. Go to tests tab (at the bottom)
4. Run it (tests are successful)
5. Check console tab (at the bottom)

---- 

In my terminal I run `jest`.

### What is expected?
Expect test not to throw a warning as all the required props are passed. The test shows that the required $props are passed. 

### What is actually happening?
Vue passes the $attrs and $listeners to the subcomponent. But whilst running the unit test, Vue throws an error that not all required $props are passed. 

---
Hi everybody, I just wanted to say I love the vue-test-utils, and I wanted to share this experience as it didn't work as I expected. I absolutely love it. 

Maybe I am testing Vue too much with this test, because you can expect the component-component to render the correct file when using it. So perhaps this test is a bit redundant. But I wanted to share my experience anyways, so you can have this background information to improve the test-utils.

My tests are actually successful (all $props are present and valid), and the component renders, but I get these errors in my console whilst running the unit test. 

```js
[Vue warn]: Missing required prop: "value"
      
      found in
      
      ---> <PriceModeCurrency>
             <Root>
```

**Comment by LinusBorg** [comment]
This is actually expected behaviour. Your problem is that you are misunderstanding how `propsData` works.

You are aware that Vue will split any attributes that you define on a child component into `$props` and `$attrs` in that child component, depending on wether or not they were defined as a prop in the child.

And you seem to expect the same to happen for the content of `propsData` - but that's not the case.

The `propsData` option (which is not not a test-utils option, but a Vue option) is used as the input for the defined props - it's applied *after* the process of splitting attributes into `$props` and $attrs woudl have taken place, so to speak.

In other words: With this `propsData`:
```javascript
propsData = {
  mode: "currency",
  value: 1000
};
```
you expect that 
1. `mode` becomes a member of `$props, since its defined as a prop of PriceMode
2. `value` becomes a member of `$attrs`, since its *not* defined as a prop of PriceMode

Your 1. is correct, 2. is false. 

vue-test-utils has a separate `attrs` topion for this reason:

https://vue-test-utils.vuejs.org/en/api/options.html#attrs

**Comment by sybrendotinga** [comment]
Thank you @LinusBorg for your time and clarifying the code! After reading this I knew what you ment. I looked at the test and added the `attrs`, and I updated the CodeSandbox. The test works. Thanks again for your detailed explanation!
